app/scripts/modules/core/src/cluster/ClusterPostSearchResultSearcher.ts

```diff
@@ -42,14 +48,51 @@ export class ClusterPostSearchResultSearcher implements IPostSearchResultSearche
       }
     }), 'cluster');
 
-    const formatter: ISearchResultFormatter = searchResultFormatterRegistry.get(ClusterPostSearchResultSearcher.TYPE);
-    return this.$q.when([{
-      id: ClusterPostSearchResultSearcher.TYPE,
-      category: ClusterPostSearchResultSearcher.TYPE,
-      icon: formatter.icon,
-      iconClass: '',
-      order: formatter.order,
-      results: clusters
-    }]);
+    const requests: IPromise<IServerGroup>[] =
```

Comment:
Requests each ASG individually as requesting all at once is very expensive (currently).  Also, this allows us to flag ASGs as to wether or not it still exists so it can be culled later.